gadget.core.Dockerfile

```diff
@@ -0,0 +1,94 @@
+# Dockerfile for Inspektor Gadget CO-RE container image gadgets.
+# This is a reduced gadget container image that supports CO-RE gadgets
+# only, i.e, it doesn't depend on BCC. This image is much smaller than
+# the traditional image and is designed to be used on systems that
+# support BTF (CONFIG_DEBUG_INFO_BTF).
+
+ARG BUILDER_IMAGE=ubuntu:20.04
+ARG BASE_IMAGE=alpine:3.14
+
+# Prepare and build gadget artifacts in a container
+FROM ${BUILDER_IMAGE} as builder
+
+ARG ENABLE_BTFGEN=false
+ENV ENABLE_BTFGEN=${ENABLE_BTFGEN}
+
+RUN set -ex; \
+	export DEBIAN_FRONTEND=noninteractive; \
+	apt-get update && \
+	apt-get install -y gcc make ca-certificates git clang \
+		software-properties-common libseccomp-dev llvm && \
+	add-apt-repository -y ppa:tuxinvader/kernel-build-tools && \
+	apt-add-repository -y ppa:longsleep/golang-backports && \
+	apt-get update && \
+	apt-get install -y libbpf-dev golang-1.17 && \
+	ln -s /usr/lib/go-1.17/bin/go /bin/go
+
+# Download BTFHub files
+COPY ./tools /btf-tools
+RUN set -ex; mkdir -p /tmp/btfs && \
+	if [ "$ENABLE_BTFGEN" = true ]; then \
+		cd /btf-tools && \
+		./getbtfhub.sh; \
+	fi
+
+# Cache go modules so they won't be downloaded at each build
+COPY go.mod go.sum /gadget/
+RUN cd /gadget && go mod download
+
+# This COPY is limited by .dockerignore
+COPY ./ /gadget
+RUN cd /gadget/gadget-container && make gadget-container-deps
+
+# Execute BTFGen
+RUN set -ex; \
+	if [ "$ENABLE_BTFGEN" = true ]; then \
+		cd /btf-tools && \
+		BTFHUB=/tmp/btfhub INSPEKTOR_GADGET=/gadget ./btfgen.sh; \
+	fi
+
+# Main gadget image
+FROM ${BASE_IMAGE}
+
+# install libseccomp according to the package manager available on the base image
```

Comment:
It is missing `rpm2cpio` and `cpio` for OpenShift.

Reply:
These are needed for downloading the kernel headers but this image is about CO-Re tools, hence those should not be needed. What happens if they are not installed?, is there a warning or the whole gadget container creation fails?

Reply:
You are right. Currently, it prints an error but it goes ahead with the container initialization. Once we have CORE implementation for all the gadgets, we can avoid downloading the kernel headers if `CONFIG_DEBUG_INFO_BTF` is set.